backend/api/contacts/src/main/java/co/airy/core/contacts/Stores.java

```diff
@@ -0,0 +1,198 @@
+                        if (contact.getDisplayName() == null || (contact.getDisplayName().equals(defaultDisplayName)
+                                && displayNameOrDefault.equals(defaultDisplayName))) {
+                            metadataList.add(newContactMetadata(contact.getId(), Contact.MetadataKeys.DISPLAY_NAME, displayNameOrDefault));
+                        }
+                        final String avatarUrl = conversation.getMetadataMap().getMetadataValue(MetadataKeys.ConversationKeys.Contact.AVATAR_URL);
+                        if (contact.getAvatarUrl() == null && avatarUrl != null) {
+                            metadataList.add(newContactMetadata(contact.getId(), Contact.MetadataKeys.AVATAR_URL, avatarUrl));
+                        }
+                    }
+
+                    return metadataList.stream()
+                            .map((metadata) -> KeyValue.pair(getId(metadata).toString(), metadata)).collect(toList());
+                })
+                .to(applicationCommunicationContacts);
+
+        streams.start(builder.build(), appId);
+    }
+
+    private ReadOnlyKeyValueStore<String, MetadataMap> getContactStore() {
+        return streams.acquireLocalStore(contactsStore);
+    }
+
+    private ReadOnlyKeyValueStore<String, Conversation> getConversationsStore() {
+        return streams.acquireLocalStore(conversationsStore);
+    }
+
+    private ReadOnlyKeyValueStore<String, MetadataMap> getConversationToContactStore() {
+        return streams.acquireLocalStore(conversationToContactStore);
+    }
+
+    public List<Contact> getAllContacts() {
+        final ReadOnlyKeyValueStore<String, MetadataMap> store = getContactStore();
+        final List<Contact> contacts = new ArrayList<>();
+        store.all().forEachRemaining(entry -> contacts.add(Contact.fromMetadataMap(entry.value)));
+        return contacts;
+    }
+
+    public Contact getContact(String contactId) {
+        final ReadOnlyKeyValueStore<String, MetadataMap> store = getContactStore();
+        return Contact.fromMetadataMap(store.get(contactId));
+    }
+
+    public Contact getContactByConversationId(String conversationId) {
+        final ReadOnlyKeyValueStore<String, MetadataMap> store = getConversationToContactStore();
+        final MetadataMap metadataMap = store.get(conversationId);
+        return Contact.fromMetadataMap(metadataMap);
+    }
+
+    public void storeContact(List<Metadata> metadataList) throws Exception {
+        for (Metadata metadata : metadataList) {
+            producer.send(new ProducerRecord<>(applicationCommunicationContacts, getId(metadata).toString(),
+                    // Interpret "" as a deletion
+                    metadata.getValue().equals("") ? null : metadata)).get();
+        }
+    }
+
+    @Override
+    public Health health() {
+        getContactStore();
+        getConversationsStore();
+        getConversationToContactStore();
+        return Health.up().build();
+    }
```

Comment:
@juan-sebastian It's true that it could be more explicit. We do it like that everywhere though and the "advantage" of this approach is that Spring automatically includes the thrown error in the status response and there's a log of course.